plugins/catalog/src/components/CatalogFilter/StarredCount.test.tsx

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { render } from '@testing-library/react';
+import { wrapInTestApp } from '@backstage/test-utils';
+import { StarredCount } from './StarredCount';
+import * as Hooks from '../../hooks/useStarredEntites';
+
+describe('Starred Count', () => {
+  it('should render the count returned from the hook', async () => {
+    jest.spyOn(Hooks, 'useStarredEntities').mockReturnValue({
```

Comment:
It's traversing boundaries a bit, but it could be nicer to just mock the storage api instead.

Also

```ts
import { MockStorage } from '@backstage/test-utils';

const storage = new MockStorage({
  'settings/starredEntities': ['id1', 'id2, 'id3, 'id4']
})
```

should be a thing :>

Reply:
Nice. I like this. Will implement. +1

Reply:
Is it possible to inject `storage` into the `useApi()` call?

Reply:
For now it's wrapping with an ApiProvider and iirc you still need to wrap the APIs in an ApiRegistry too, wanna clean that up a bit in not too long though.

Reply:
Oh the test app created by test-utils currently has an empty ApiTestRegistry in the app. The plan is to make that contain a bunch of mocks by default, and allow direct access to those mocks too. The mocks would be set up to reset between every test as well.

Reply:
Will leave this for now until we do Starred Items